src/RE.h

```diff
@@ -96,6 +96,11 @@ class Specific_RE_Matcher {
     // to the matching expressions.  (idx must not contain zeros).
     bool CompileSet(const string_list& set, const int_list& idx);
 
+    // For use with CompileSet() to collect indices of all matched
+    // expressions into the matches vector. Behaves as MatchAll(),
+    // consuming the complete input string.
```

Comment:
Might be good to expand that comment a bit since the `matches` result argument isn't something that's in the other (public) `MatchAll()` implementations.